app/services/submit_reference.rb

```diff
@@ -18,11 +18,41 @@ def save!
     if @send_emails
       CandidateMailer.reference_received(reference).deliver_later
       RefereeMailer.reference_confirmation_email(application_form, reference).deliver_later
+      notify_provider_users
     end
   end
 
 private
 
+  def notify_provider_users
+    return unless FeatureFlag.active?(:new_references_flow_providers)
+
+    application_choices.each do |ac|
+      ac.provider.provider_users.each do |pu|
+        ProviderMailer.reference_received(
+          provider_user: pu,
+          application_choice: ac,
+          reference: reference,
+          course: course,
+          ordinance: ordinance,
+        ).deliver_later
+      end
+    end
+  end
+
+  # i.e 1st, 2nd, 3rd
+  def ordinance
+    @ordinance ||= TextOrdinalizer.new(reference.order_in_application_references).call
+  end
+
+  def course
+    @course ||= reference.candidate.course_from_find
+  end
```

Comment:
Any reason not to look up the course from the application_choice instead? I'm not entirely sure what `course_from_find` is?